modules/solid_mechanics/src/bcs/InteractionIntegralBenchmarkBC.C

```diff
@@ -32,34 +32,43 @@ InteractionIntegralBenchmarkBC::InteractionIntegralBenchmarkBC(const InputParame
     _crack_front_point_index(getParam<unsigned int>("crack_front_point_index")),
     _poissons_ratio(getParam<Real>("poissons_ratio")),
     _youngs_modulus(getParam<Real>("youngs_modulus")),
-    _ki(getParam<Real>("KI")),
-    _kii(getParam<Real>("KII")),
-    _kiii(getParam<Real>("KIII"))
+    _ki_function(&getFunction("KI_function")),
+    _kii_function(&getFunction("KII_function")),
+    _kiii_function(&getFunction("KIII_function"))
 {
-  _kappa = 3 - 4*_poissons_ratio;
-  _mu = _youngs_modulus / (2*(1 + _poissons_ratio));
+  _kappa = 3 - 4 * _poissons_ratio;
+  _mu = _youngs_modulus / (2 * (1 + _poissons_ratio));
 }
 
 Real
 InteractionIntegralBenchmarkBC::computeQpValue()
 {
+  Real ki_val;
+  ki_val = _ki_function->value(_t, *_current_node);
```

Comment:
Not a big deal but you may as well combine these two lines and the similar lines below.